Lib/test/test_dict.py

```diff
@@ -1312,6 +1312,19 @@ def test_reversed(self):
         self.assertEqual(list(r), list('dcba'))
         self.assertRaises(StopIteration, next, r)
 
+    def test_reverse_iterator_for_empty_dict(self):
+        empty_dict = {}
+        self.assertEqual(list(reversed(empty_dict)), [])
+        self.assertEqual(list(reversed(empty_dict.items())), [])
+        self.assertEqual(list(reversed(empty_dict.values())), [])
+        self.assertEqual(list(reversed(empty_dict.keys())), [])
+
+        empty_dict = dict()
```

Comment:
These tests look redundant.

Reply:
They are not redudant because `dict` and `{}` don't trigger the same code path.

Indeed, the error is not reproducible by

`list(reversed(dict))`

but only for

`list(reversed({}))`

Reply:
@pablogsal is correct. I will re-add dict() case.